extractors-canopycover/drone_canopycover.py

```diff
@@ -0,0 +1,715 @@
+MAX_FILE_OPEN_SLEEP_SEC = 30
+
+# Array of trait names that should have array values associated with them
+TRAIT_NAME_ARRAY_VALUE = ['canopy_cover', 'site']
+
+# Mapping of default trait names to fixecd values
+TRAIT_NAME_MAP = {
+    'access_level': '2',
+    'species': 'Unknown',
+    'citation_author': '"Zongyang, Li"',
+    'citation_year': '2016',
+    'citation_title': 'Maricopa Field Station Data and Metadata',
+    'method': 'Canopy Cover Estimation from RGB images'
+}
+
+# Used to generate random nuymbers
+RANDOM_GENERATOR = None
+
+def update_geostreams(connector, host, secret_key, geo_csv_header, geo_rows):
+    """Sends the rows of csv data to geostreams
+    Args:
+        connector(obj): the message queue connector instance
+        host(str): the URI of the host making the connection
+        secret_key(str): used with the host API
+        geo_csv_header(str): comma separated list of column headers
+        geo_rows(list): list of strings that are comma separated column data (list of data rows)
+    Notes:
+        Header names expected are: 'lat', 'lon', 'dp_time', 'timestamp', 'source', 'value', and 'trait'
+    """
+    data = [geo_csv_header]
+    data.extend(geo_rows)
+
+    reader = csv.DictReader(data)
+    idx = 1
+    for row in reader:
+        centroid_lonlat = [row['lon'], row['lat']]
+        time_fmt = row['dp_time']
+        timestamp = row['timestamp']
+        dpmetadata = {
+            "source": row['source'],
+            "value": row['value']
+        }
+        trait = row['trait']
+
+        idx += 1
+        create_datapoint_with_dependencies(connector, host, secret_key, trait,
+                                           (centroid_lonlat[1], centroid_lonlat[0]), time_fmt, time_fmt,
+                                           dpmetadata, timestamp)
+
+def update_betydb(bety_csv_header, bety_rows):
+    """Sends the rows of csv data to BETYdb
+    Args:
+        bety_csv_header(str): comma separated list of column headers
+        bety_rows(list): list of strings that are comma separated column data (list of data rows)
+    """
+    betyurl = get_bety_api('traits')
+    request_params = {'key': get_bety_key()}
+    filetype = 'csv'
+    content_type = 'text/csv'
+    data = [bety_csv_header]
+    data.extend(bety_rows)
+
+    resp = requests.post("%s.%s" % (betyurl, filetype), params=request_params,
+                         data=os.linesep.join(data),
+                         headers={'Content-type': content_type})
+
+    if resp.status_code in [200, 201]:
+        logging.info("Data successfully submitted to BETYdb.")
+        return resp.json()['data']['ids_of_new_traits']
+    else:
+        logging.error("Error submitting data to BETYdb: %s -- %s", resp.status_code, resp.reason)
+        resp.raise_for_status()
+
+    return None
+
+def _get_plot_name(name):
+    """Looks in the parameter and returns a plot name.
+
+       Expects the plot name to be identified by having "By Plot" embedded in the name.
```

Comment:
I don't understand. Does the string 'by plot' have to be in the name?

Reply:
Yes it does. That's how plot naming is handled in TERRA REF at this time

Reply:
where? in BETYdb, plot names are of the form `MAC Field Scanner Season 4 Range 51 Column 7`

Reply:
I put a ticket in to store the plot name in the metadata so that the dataset naming can be ignored